pkg/proxy/proxy.go

```diff
@@ -21,18 +21,18 @@ import (
 	"crypto/tls"
 	"crypto/x509"
 	"fmt"
-	"io/ioutil"
 	"net/http"
 	"net/http/httputil"
 	"net/url"
+	"os"
 
 	"k8s.io/apimachinery/pkg/util/runtime"
 	userinfo "k8s.io/apiserver/pkg/authentication/user"
 	"k8s.io/apiserver/pkg/endpoints/request"
 )
 
 func newTransport(clientCert, clientKeyFile, caFile string) (*http.Transport, error) {
-	caCert, err := ioutil.ReadFile(caFile)
+	caCert, err := os.ReadFile(caFile)
```

Comment:
Is this a Go 1.19 thing? Is it valid under 1.18?

Reply:
https://pkg.go.dev/io/ioutil

> Deprecated: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details.

should be valid up until 1.16

Reply:
ACK

Reply:
I'm using 1.19, but it should be valid under 1.18. ioutil has been deprecated since 1.16 according to the docs https://pkg.go.dev/io/ioutil